packages/tailwindcss/src/index.ts

```diff
@@ -0,0 +1,77 @@
+import { FormKitNode, FormKitClasses } from '@formkit/core'
+import plugin from 'tailwindcss/plugin'
+
+// FormKit plugin for Tailwind
+// used in tailwind.config.js
+const formKitVariants = plugin(({ addVariant }) => {
+  addVariant('formkit-disabled', ['&[data-disabled]', '[data-disabled] &', '[data-disabled]&'])
+  addVariant('formkit-invalid', ['&[data-invalid]', '[data-invalid] &', '[data-invalid]&'])
+  addVariant('formkit-errors', ['&[data-errors]', '[data-errors] &', '[data-errors]&'])
+  addVariant('formkit-complete', ['&[data-complete]', '[data-complete] &', '[data-complete]&'])
+  addVariant('formkit-loading', ['&[data-loading]', '[data-loading] &', '[data-loading]&'])
+  addVariant('formkit-submitted', ['&[data-submitted]', '[data-submitted] &', '[data-submitted]&'])
+  addVariant('formkit-multiple', ['&[data-multiple]', '[data-multiple] &', '[data-multiple]&'])
+})
+
+// Helper function for Tailwind styling.
+// to be imported and used in FormKit config
+export function generateClasses (
```

Comment:
@justin-schroeder This function has the desired behavior, can create a theme from a POJO structured as 
```
{
  [type]: {
    [sectionKey]: '',
    [sectionKey]: '',
    ...
  }
  ...
}
```
I'm going to need your help cutting my teeth on the proper TS definitions / return types. Had to cop out and use `any` for now and then i'm accessing a few function arguments in`formKitStates` because I don't need them but I can't _not_ use them.

Reply:
I had a go with this here - hopefully this should help!
```ts
import type { DefaultConfigOptions, FormKit } from '@formkit/vue'
import type { inputs } from "@formkit/inputs"

type InputTypes = keyof typeof inputs | "global"

// https://formkit.com/essentials/inputs#sections
type SectionKeys = "outer" | "wrapper" | "label" | "prefix" | "inner" | "suffix" | "input" | "help" | "messages" | "message"

type classConfigSchema = Partial<{
    [key in InputTypes]: Partial<{
        [key in SectionKeys]: (() => string) | string;
    }>
}>
```